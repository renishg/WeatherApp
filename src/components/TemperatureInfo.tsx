import React, {FC} from 'react';
import {View, Text, StyleSheet, Image} from 'react-native';
import {colors} from '../theme/colors';
import {fonts} from '../theme/fonts';

type TemperatureInfoProps = {
  tempC: number;
  feelsLikeC: number;
  conditionText: string;
  conditionIcon: string;
};

const TemperatureInfo: FC<TemperatureInfoProps> = ({
  tempC,
  feelsLikeC,
  conditionText,
  conditionIcon,
}) => {
  return (
    <View style={styles.container}>
      <View style={styles.tempContainer}>
        <Text style={styles.tempText}>{Math.round(tempC)}</Text>
        <Text style={styles.tempUnitText}>°C</Text>
        <Text style={styles.feelsLikeText}>{`Feels like ${feelsLikeC}°C`}</Text>
      </View>
      <View
        style={StyleSheet.flatten([
          styles.tempContainer,
          styles.conditionContainer,
        ])}>
        <Image
          source={{uri: `https:${conditionIcon}`}}
          style={styles.conditionIcon}
          resizeMode={'contain'}
        />
        <Text style={styles.conditionText}>{conditionText}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
  },
  tempContainer: {
    flexDirection: 'row',
  },
  tempText: {
    color: colors.white,
    fontSize: 100,
    fontFamily: fonts.semiBold,
  },
  tempUnitText: {
    color: colors.white,
    fontSize: 40,
    marginTop: 30,
    fontFamily: fonts.medium,
  },
  conditionText: {
    color: colors.white,
    fontSize: 30,
    fontFamily: fonts.medium,
  },
  feelsLikeText: {
    color: colors.white,
    fontSize: 18,
    fontFamily: fonts.light,
    alignSelf: 'flex-end',
    marginBottom: 30,
    marginLeft: -30,
  },
  conditionIcon: {
    width: 50,
    height: 50,
    marginRight: 10,
  },
  conditionContainer: {
    alignItems: 'center',
    marginBottom: 24,
  },
});

export {TemperatureInfo};
