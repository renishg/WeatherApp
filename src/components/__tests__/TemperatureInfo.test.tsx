import React from 'react';
import {create} from 'react-test-renderer';
import {Image} from 'react-native';
import {TemperatureInfo} from '../TemperatureInfo';

describe('TemperatureInfo', () => {
  const mockProps = {
    tempC: 30,
    feelsLikeC: 32,
    conditionText: 'Sunny',
    conditionIcon: 'https://www.example.com/icon.png',
  };

  it('should render', () => {
    const tree = create(<TemperatureInfo {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the correct condition icon', () => {
    const tree = create(<TemperatureInfo {...mockProps} />);
    const conditionIcon = tree.root.findByType(Image);

    expect(conditionIcon.props.source.uri).toBe(
      `https:${mockProps.conditionIcon}`,
    );
  });
});
