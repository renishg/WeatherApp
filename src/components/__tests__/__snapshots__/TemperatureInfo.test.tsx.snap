// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureInfo should render 1`] = `
<View
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-SemiBold",
          "fontSize": 100,
        }
      }
    >
      30
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Medium",
          "fontSize": 40,
          "marginTop": 30,
        }
      }
    >
      °C
    </Text>
    <Text
      style={
        {
          "alignSelf": "flex-end",
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Light",
          "fontSize": 18,
          "marginBottom": 30,
          "marginLeft": -30,
        }
      }
    >
      Feels like 32°C
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "https:https://www.example.com/icon.png",
        }
      }
      style={
        {
          "height": 50,
          "marginRight": 10,
          "width": 50,
        }
      }
    />
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Medium",
          "fontSize": 30,
        }
      }
    >
      Sunny
    </Text>
  </View>
</View>
`;
